src/doc/rustdoc/src/lints.md

```diff
@@ -374,3 +374,36 @@ warning: this URL is not a hyperlink
 
 warning: 2 warnings emitted
 ```
+
+## `synthetic_non_exhaustive`
```

Comment:
It's not clear what this lint does based on its name. Perhaps a name along the lines of `exhaustive_but_hidden` would be better?